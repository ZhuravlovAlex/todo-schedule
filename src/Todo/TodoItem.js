import React, { useContext } from "react";
import PropTypes from "prop-types"
import Context from "../context";
import styles from "./TodoItem.module.css"


function TodoItem({ todo, index, onChange }) {
	const { removeTodo } = useContext(Context)
	const classes = []

	if (todo.completed) {
		classes.push('done')
	}

	return (
		<li className={styles.item} >
			<span className={classes.join(' ')}>
			<strong className={styles.posNum}>{index + 1}</strong>
				&nbsp;
				<input 
					className={styles.itemInput}
					type="checkbox" 
					style={styles.input} 
					onChange={() => onChange(todo.id)} 
					checked={todo.completed}
				/>
				
				{todo.title}
			</span>

			<button className={styles.deleteTodo} 
				onClick={removeTodo.bind(null, todo.id)}>
				&times;
			</button>
		</li>
	)
}

TodoItem.propTypes = {
	todo: PropTypes.object.isRequired,
	index: PropTypes.number,
	onChange: PropTypes.func.isRequired
}

export default TodoItem