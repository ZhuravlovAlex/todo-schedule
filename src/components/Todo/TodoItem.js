import React from "react";
import PropTypes from "prop-types"
import styles from "./TodoItem.module.css"

function TodoItem({ todo, index, onChange }) {
	const classes = []

	if (todo.completed) {
		classes.push('done')
	}

	return (
		<li className={styles.item} >
			<span className={classes.join(' ')}>
				<strong className={styles.posNum}>{todo.id}</strong>
				&nbsp;
				<input
					className={styles.itemInput}
					type="checkbox"
					style={styles.input}
					onChange={() => onChange(todo.id)}
					checked={todo.completed}
				/>

				{todo.title}
			</span>

			<button className={styles.deleteTodo}>
				&times;
			</button>
		</li>
	)
}

TodoItem.propTypes = {
	todo: PropTypes.object.isRequired,
	index: PropTypes.number,
	onChange: PropTypes.func.isRequired
}

export default TodoItem;