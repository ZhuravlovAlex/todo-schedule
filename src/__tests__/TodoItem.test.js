import renderer from 'react-test-renderer';
import TodoItem from '../components/Todo/TodoItem';

const onChange = () => {

};
const todo = {
	id: 1,
	title: test,
	completed: false
}

it('renders correctly', () => {
	const tree = renderer.create(<TodoItem
		todo={todo}
		onChange={onChange}
	/>).toJSON();

	expect(tree).toMatchSnapshot();
});


