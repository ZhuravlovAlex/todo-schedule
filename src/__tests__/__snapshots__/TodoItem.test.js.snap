// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<li
  className="item"
>
  <span
    className=""
  >
    <strong
      className="posNum"
    >
      1
    </strong>
     
    <input
      checked={false}
      className="itemInput"
      onChange={[Function]}
      style="input"
      type="checkbox"
    />
  </span>
  <button
    className="deleteTodo"
  >
    ×
  </button>
</li>
`;
